### PR TITLE
Fix multiple timer issue in CopyButton by using useRef

### PR DESCRIPTION
**Pull request by sohzm**
This commit resolves a bug where rapidly clicking the Copy button in the CopyButton component would initiate multiple overlapping timers, leading to unpredictable copy state toggling. By utilizing useRef, we now ensure a single timer instance is managed and cleared appropriately, stabilizing the copy functionality.

#### Summary
Earlier we didnt clear the old timer before creating another one when copy button was pressed, this pr fixes that.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/copy_button.tsx [hunk 1]
```diff
@@ -22,12 +22,18 @@ type Props = {
...
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
 
     const copyText = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void => {
         e.preventDefault();
         setIsCopied(true);
 
-        setTimeout(() => {
+        if (timerRef.current) {
+            clearTimeout(timerRef.current);
+            timerRef.current = null;
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/copy_button.tsx:33]: we probably dont need to assign it to null because we are assigning it in the next line

Comment (sohzm) [reply webapp/channels/src/components/copy_button.tsx]: yes that sounds good, Ive done the change

**Comment by M-ZubairAhmed** [comment]
@sohzm just a request, would you be open to making this component not use the react bootstrap tooltip and use our inbuilt useTooltip (floating-ui) , just a thought. feel free to decline if you dont want to 😄

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed should that be done in a seperate pr?

Yes you could do that if you want to. i will approve this pr

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Thanks for the PR, @sohzm! That's a great catch.

**Pull request merged**